src/enum_net.cpp

```diff
@@ -168,19 +168,24 @@ namespace {
 
 #if TORRENT_USE_NETLINK
 
-	int read_nl_sock(int sock, span<char> buf, std::uint32_t const seq, std::uint32_t const pid)
+	int read_nl_sock(int sock, std::vector<char>& buf, std::uint32_t const seq, std::uint32_t const pid)
 	{
 		nlmsghdr* nl_hdr;
 
-		int msg_len = 0;
-
+		int increment = 1024;
+		int const max_increment = 8192;
+		int read_cursor = 0;
 		for (;;)
 		{
-			auto next_msg = buf.subspan(msg_len);
-			int const read_len = int(recv(sock, next_msg.data(), static_cast<std::size_t>(next_msg.size()), 0));
+			if (read_cursor + (increment / 2) > int(buf.size()))
+			{
+				buf.resize(buf.size() + increment);
```

Comment:
this is the actual fix. instead of taking a `span` to a fixed size buffer, take a `std::vector` and resize it on demand.